File: flightbooker.client/src/models/DTOs/RouteBookingDataDTO.ts
export type RouteBookingDataDTO = {
    id: number;
    airlineName: string;
    departureAirportCode: string;
    arrivalAirportCode: string;
    departureTime: Date;
    arrivalTime: Date;
    flights: FlightBookingDataDTO[];
    prices: {
        economy: number;
        premiumEconomy: number;
        business: number;
        firstClass: number;
    };
};

export type FlightBookingDataDTO = {
    flightId: number;
    airplaneId: number;
    departureAirportCode: string;
    arrivalAirportCode: string;
    departureTime: Date;
    arrivalTime: Date;
    duration: number;
    delay: number;
    seats: FlightSeatDataDTO[];
};

type FlightSeatDataDTO = {
    flightSeatId: number;
    airplaneSeatId: number;
    isReserved: boolean;
}

File: FlightBooker.Server/Data/Models/DTOs/Routes/RouteBookingDataDTO.cs
namespace FlightBooker.Server.Data.Models.DTOs.Routes
{
    public class RouteBookingDataDTO
    {
        public int Id { get; set; }
        public string AirlineName { get; set; } = null!;
        public string DepartureAirportCode { get; set; } = null!;
        public string ArrivalAirportCode { get; set; } = null!;
        public DateTime DepartureTime { get; set; }
        public DateTime ArrivalTime { get; set; }
        public ICollection<FlightBookingDataDTO> Flights { get; set; } = [];
        public Dictionary<string, decimal> Prices { get; set; } = [];
    }

    public class FlightBookingDataDTO
    {
        public int FlightId { get; set; }
        public string DepartureAirportCode { get; set; } = null!;
        public string ArrivalAirportCode { get; set; } = null!;
        public DateTime DepartureTime { get; set; }
        public DateTime ArrivalTime { get; set; }
        public int Duration { get; set; }
        public int? Delay { get; set; } = null;
        public int AirplaneId { get; set; }
        public ICollection<FlightSeatDataDTO> Seats { get; set; } = [];
    }

    public class FlightSeatDataDTO
    {
        public int FlightSeatId { get; set; }
        public int AirplaneSeatId { get; set; }
        public bool IsReserved { get; set; } = false;
    }
}
